pkg/status/combinedstatus-resolver.go

```diff
@@ -468,6 +507,41 @@ func (c *combinedStatusResolver) evaluateWorkStatusesPerBindingReadLocked(bindin
 	return combinedStatusesToQueue
 }
 
+// getObjectMetaAndSpec fetches the metadata and spec of the object associated
+// with the given workload object identifier.
+// The function is guaranteed not to return a key of `status` in the map.
+// If the resource contains any other subresources, they are fetched as well.
+func (c *combinedStatusResolver) getObjectMetaAndSpec(objectIdentifier util.ObjectIdentifier) (map[string]interface{}, error) {
+	// fetch object
+	lister, exists := c.wdsListers.Get(objectIdentifier.GVR())
+	if !exists {
+		return nil, fmt.Errorf("lister not found for gvr %s", objectIdentifier.GVR())
+	}
+
+	obj, err := lister.ByNamespace(objectIdentifier.ObjectName.Namespace).Get(objectIdentifier.ObjectName.Name)
```

Comment:
What if the resource is not namespaced?

Reply:
Cluster-wide resources do not make it this far, but yes this function should probably be agnostic of that.

On the first note: since an object cannot be both namespace scoped and cluster scoped, CombinedStatus currently supports namespace-scoped objects only. We can duplicate for cluster-scoped resources and fit them through the same pipeline later.

Reply:
What do you mean by "CombinedStatus currently supports namespace-scoped objects only"?

See https://github.com/kubestellar/kubestellar/blob/70ded98849b7236c5c160667c55c8b3f943b39d8/api/control/v1alpha1/types.go#L447

Reply:
I was wondering how this didnt bother me until now, thanks, forgot that. Will fix.

Reply:
Fixed.